### ISSUE TITLE
The `input_ids` and `labels` information during fine-tuning

### ISSUE THREAD

**Issue by sykverse** [openvla/openvla#134]
Hi,
Thank you for sharing your great work.
I have a question about the` input_ids` and `labels` information during fine-tuning.

It seems that when fine-tuning the model using `finetune.py`, the `labels` information is included in the `input_ids`.
Furthermore, it appears that the training proceeds without any masking process for the `labels` information in the `input_ids`.
I believe that training the model with the `labels` information included in the `input_ids` could be considered a form of data leakage. 
I would like to ask if there is a specific reason for structuring the code in this way.

```
# dataset.py

# example dataset: Open X Embodiment - austin_buds_dataset_converted_externally_to_rlds
# print(prompt_builder.get_prompt()) -> 'In: What action should the robot take to opening the fridge?\nOut: ပ右删ἘြぶŸ</s>'
input_ids = self.base_tokenizer(prompt_builder.get_prompt(), add_special_tokens=True).input_ids
labels = list(input_ids)

# Tensorize =>> Run Image Transform to get `pixel_values` =>> Return
#   =>> IMPORTANT :: IF WE'RE USING HF LLM.forward(..., labels=labels), SHIFTING HAPPENS _INSIDE_ MODEL!
input_ids, labels = torch.tensor(input_ids), torch.tensor(labels)
pixel_values = self.image_transform(img)

# [CRITICAL] We do not want to take the loss for anything but the predicted action tokens!
labels[: -(len(action) + 1)] = IGNORE_INDEX
if not self.predict_stop_token:
    labels[-1] = IGNORE_INDEX
```

**Comment by kpertsch** [comment]
The way autoregressive sequence models are typically trained is by taking all tokens T[0:t-1] as conditioning and predicting the next token T[t]. So the way this is implemented is by putting all tokens into a long sequence of [inputs, outputs] and then during model training feeding the full sequence into the transformer (with causal attention masking) and computing loss towards the *same* sequence right-shifted by one as the target, with appropriate loss masking. Note the comment that the right shifting happens inside the model, so we don't need to right-shift the labels outselves.
So this design of putting inputs and outputs into one sequence and running the model on all of them is pretty standard.
